subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/scheduling/TestsSchedulerFactory.kt

```diff
@@ -15,56 +13,28 @@ import com.avito.instrumentation.internal.suite.filter.FilterInfoWriter
 import com.avito.instrumentation.metrics.InstrumentationMetricsSender
 import com.avito.retrace.ProguardRetracer
 import com.avito.runner.service.worker.device.adb.listener.RunnerMetricsConfig
-import com.avito.time.DefaultTimeProvider
-import com.google.common.annotations.VisibleForTesting
+import com.avito.time.TimeProvider
 import com.google.gson.Gson
+import java.io.File
+import java.nio.file.Files
 
 internal interface TestsSchedulerFactory {
 
-    fun create(): TestsScheduler
+    fun create(devicesProviderFactory: DevicesProviderFactory): TestsScheduler
```

Comment:
Can't we hide it in implementation class?

Reply:
i'm afraid not in this moment of refactoring, i exposed it only to be able to provide it both for old action and new service ways